rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -363,6 +367,58 @@ public CoordinateBuilder.Expression getCoordinates() {
         }
     }
 
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @Data
+    final class ArrayTypeTree implements J, TypeTree, Expression {
+        @With
+        @EqualsAndHashCode.Include
+        UUID id;
+
+        @With
+        Space prefix;
+
+        @With
+        Markers markers;
+
+        @With
+        TypeTree elementType;
+
+        @Nullable
+        @With
+        List<J.Annotation> annotations;
+
+        @With
+        JRightPadded<Space> dimension;
+
+        @Override
+        public JavaType getType() {
+            return elementType.getType();
```

Comment:
Hmm. I can see that this is also what was done for `J.ArrayType`, but this strikes me as wrong. I think the method should return a type attribution representing itself, not the element type. I am unsure now where this method would end up getting called (to be analyzed), but I still think this is probably wrong.

Reply:
I had a similar thought. I thought it may help to add a `getArrayType`, similar to `getVariableType` or `getMethodType`, but I wasn't sure.

Reply:
I agree with Knut's original suggestion here and change the type to be an array type.

I think long ago we didn't have `getVariableType`, etc. but only a `getType`, so we were choosing between these sometimes. Only much later did we resolve that tension by adding these extra methods. It would be a mistake in this case to add a `getArrayType`. If anything (and I don't feel it necessary right now) you would do the opposite and add a `getElementType()` that returns the element type while changing `getType()` to return the array type.

Reply:
I see, that does make more sense, since the type is a `JavaType.Array` and `getType` currently returns the `elementType` instead. At the moment our plan is to reuse the `J.ArrayType`. I'll add a `JavaType.Array arrayType` field and update `getType`.

Reply:
We should just call the field `type` and then won't need explicit accessors.